### PR TITLE
CI:  Fix GHA failure notification trigger condition

### PR DESCRIPTION
**Pull request by michaeldiamant**
Amends the trigger condition for Github Action (GHA) failure notification so that it _only_ reaches out when the branch being built matches `master` or a release branch.

From the docs in https://docs.github.com/en/actions/learn-github-actions/contexts:

Property name | Type | Description
-- | -- | --
github.base_ref | string | The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.
github.ref_name | string | The short ref name of the branch or tag that triggered the workflow run. This value matches the branch or tag name shown on GitHub. For example, feature-branch-1.

Details:
* I'm not sure of the original intent in https://github.com/algorand/go-algorand/pull/4791/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L48, but based on the docs, I think the desired behavior is `ref_name`.
* https://github.com/algorand/go-algorand/commit/a275d96d66720f6747485632265ad464f2567135 is tested in https://github.com/algorand/go-algorand/pull/4808 (produces no notification) and cherry-picked into this PR.

### CODE REVIEW

**Comment by algojack** [comment]
@michaeldiamant can we clarify here are we trying to have notifications for PRs that
1. target(base) master/rel* branches or
2. the ones that trigger a build after merging
i think with this PR we are switching from 1 to 2. Please correct me if i misunderstand the description.

**Review by algolucky — APPROVED** [review]

File: .github/workflows/benchmarks.yml [hunk 1]
```diff
@@ -54,4 +54,4 @@ jobs:
           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
         run: |
           curl -X POST --data-urlencode "payload={\"text\": \"Benchmark workflow failed. ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }} \"}" $SLACK_WEBHOOK
-        if: ${{ failure() && (contains(github.base_ref, 'rel/nightly') || contains(github.base_ref, 'rel/beta') || contains(github.base_ref, 'rel/stable') || contains(github.base_ref, 'master')) }}
+        if: ${{ failure() && (contains(github.ref_name, 'rel/nightly') || contains(github.ref_name, 'rel/beta') || contains(github.ref_name, 'rel/stable') || contains(github.ref_name, 'master')) }}
```
Comment (algojack) [inline .github/workflows/benchmarks.yml:57]: this runs only on master, see lines 6 and 10

**Comment by michaeldiamant** [comment]
> I think with this PR we are switching from 1 to 2.

@algojack Correct.  That's what I had intended to do in #4791.

File: .github/workflows/benchmarks.yml (hunk 1 above)
Comment (michaeldiamant) [reply .github/workflows/benchmarks.yml:57]: Agree - I kept as is to minimize issues arising from copy-paste among ymls.

**Comment by algojack** [comment]
Seems like this is intentional. so LGTM.

**Review by algojack — APPROVED** [review]

**Review by algobarb — APPROVED** [review]
I think this change is correct to change from `base_ref` to `ref_name`. 

It might be a separate discussion on whether we only want to notify on failures these particular branches.

**Review by algobarb — APPROVED** [review]

**Pull request merged**